linetools/lists/linelist.py

```diff
@@ -341,11 +345,16 @@ def set_extra_columns_to_datatable(self, abundance_type='solar', ion_correction=
             Ionization correction. Options are:
                 'none' : No correction applied, so this column will
                          be filled with zeros. (Default)
+         redo : bool, optional
+            Remake the extra columns
 
         Note: This function is only implemented for the following
         lists: HI, ISM, EUV, Strong.
 
         """
+        # Avoid redo (especially for caching)
+        if ('ion_name' in self._data.keys()) and (not redo):
+            return
 
```

Comment:
Looks like a good solution for now, but I think this should be controled at the LineList.__init__() level. I am envisioning this method to set different kinds of abundances, and ionization correction options as given in the initialization of LineList.

An alternative then would be to add a keyword attribute in .__init__() like `set_extra_columns=False` or something like that...